mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/task/RecomputeCurrentQuotasServiceContract.java

```diff
@@ -0,0 +1,143 @@
+
+    @Test
+    default void recomputeCurrentQuotasShouldReturnCompleteWhenUserWithNoMessage() throws Exception {
+        usersRepository().addUser(USER_1, PASSWORD);
+
+        assertThat(testee().recomputeCurrentQuotas(new Context()))
+            .isEqualTo(Task.Result.COMPLETED);
+    }
+
+    @Test
+    default void recomputeCurrentQuotasShouldComputeEmptyQuotasWhenUserWithNoMessage() throws Exception {
+        usersRepository().addUser(USER_1, PASSWORD);
+
+        testee().recomputeCurrentQuotas(new Context());
+
+        assertThat(currentQuotaManager().getCurrentQuotas(userQuotaRootResolver().forUser(USER_1)))
+            .isEqualTo(CurrentQuotas.emptyQuotas());
+    }
+
+    @Test
+    default void recomputeCurrentQuotasShouldReturnCompleteWhenUserWithMessage() throws Exception {
+        usersRepository().addUser(USER_1, PASSWORD);
+
+        MailboxSession session = sessionProvider().createSystemSession(USER_1);
+        mailboxManager().createMailbox(MAILBOX_PATH, session);
+
+        MessageManager messageManager = mailboxManager().getMailbox(MAILBOX_PATH, session);
+        appendMessageToUser(messageManager, session);
+
+        assertThat(testee().recomputeCurrentQuotas(new RecomputeCurrentQuotasService.Context()))
+            .isEqualTo(Task.Result.COMPLETED);
+    }
+
+    @Test
+    default void recomputeCurrentQuotasShouldRecomputeCurrentQuotasCorrectlyWhenUserWithMessage() throws Exception {
+        usersRepository().addUser(USER_1, PASSWORD);
+
+        MailboxSession session = sessionProvider().createSystemSession(USER_1);
+        mailboxManager().createMailbox(MAILBOX_PATH, session);
+
+        MessageManager messageManager = mailboxManager().getMailbox(MAILBOX_PATH, session);
+        appendMessageToUser(messageManager, session);
+
+        testee().recomputeCurrentQuotas(new RecomputeCurrentQuotasService.Context());
+
+        assertThat(currentQuotaManager().getCurrentQuotas(userQuotaRootResolver().forUser(USER_1)))
+            .isEqualTo(EXPECTED_QUOTAS);
+    }
+
+    @Test
+    default void recomputeCurrentQuotasShouldResetCurrentQuotasWhenIncorrectQuotas() throws Exception {
+        usersRepository().addUser(USER_1, PASSWORD);
+
+        MailboxSession session = sessionProvider().createSystemSession(USER_1);
+        mailboxManager().createMailbox(MAILBOX_PATH, session);
+
+        MessageManager messageManager = mailboxManager().getMailbox(MAILBOX_PATH, session);
+        appendMessageToUser(messageManager, session);
+
+        QuotaRoot quotaRoot = userQuotaRootResolver().forUser(USER_1);
+
+        QuotaOperation operation = new QuotaOperation(quotaRoot, QuotaCountUsage.count(3l), QuotaSizeUsage.size(390L));
+        currentQuotaManager().increase(operation);
+
+        testee().recomputeCurrentQuotas(new RecomputeCurrentQuotasService.Context());
+
+        assertThat(currentQuotaManager().getCurrentQuotas(userQuotaRootResolver().forUser(USER_1)))
+            .isEqualTo(EXPECTED_QUOTAS);
+    }
+
```

Comment:
Can you show that it reports failure by injecting a failure, for instance in that cassandra test suite only using cassandra session scenario ?